plugins/common/cookie/cookie_test.go

```diff
@@ -232,6 +232,7 @@ func TestAuthConfig_Start(t *testing.T) {
 			srv.checkResp(t, tt.firstHTTPResponse)
 			mock.Add(renewalCheck)
 			// Ensure that the auth renewal goroutine has completed
+			time.Sleep(time.Second)
```

Comment:
Can't you check for the auth to be finished e.g. by counting the number of auth responses or similar and check with [require.Eventually](https://pkg.go.dev/github.com/stretchr/testify/require#Eventually)?

Reply:
Good suggestion, I didn't know about require.Eventually! I've updated it let me know what you think.